src/prefect/new_flow_engine.py

```diff
@@ -282,16 +280,20 @@ def enter_run_context_sync(self, client: Optional[PrefectClient] = None):
         except AsyncLibraryNotFoundError:
             task_group = anyio._backends._asyncio.TaskGroup()
 
-        with FlowRunContext(
-            flow=self.flow,
-            log_prints=self.flow.log_prints or False,
-            flow_run=self.flow_run,
-            parameters=self.parameters,
-            client=client,
-            background_tasks=task_group,
-            result_factory=run_sync(ResultFactory.from_flow(self.flow)),
-            task_runner=self.flow.task_runner,
-        ):
+        with ExitStack() as stack:
+            task_runner = stack.enter_context(ConcurrentTaskRunner())
```

Comment:
why is this hardcoded?

Reply:
Just for testing. It was the most convenient place to inject the new task runner implementation. I'm going to do another pass to clean stuff up like this before marking this PR as ready.

Reply:
👍